server/handle.go

```diff
@@ -175,10 +171,21 @@ func (l *listener4) HandleMsg4(buf []byte, oob *ipv4.ControlMessage, _peer net.A
 				log.Errorf("HandleMsg4: Did not receive interface information")
 			}
 		}
-		if _, err := l.WriteTo(resp.ToBytes(), woob, peer); err != nil {
-			log.Printf("MainHandler4: conn.Write to %v failed: %v", peer, err)
-		}
 
+		if useEthernet {
+			intf, err := net.InterfaceByIndex(woob.IfIndex)
+			if err != nil {
+				log.Errorf("MainHandler4: Can not get Interface for index %d %v", woob.IfIndex, err)
```

Comment:
Need to return from the function otherwise there will be a nil pointer dereference on l180
```suggestion
				log.Errorf("MainHandler4: Can not get Interface for index %d %v", woob.IfIndex, err)
				return
```